managed/ui/src/components/universes/UniverseForm/ClusterFields.js

```diff
@@ -492,16 +493,24 @@ export default class ClusterFields extends Component {
       this.props.fetchUniverseResources(nextProps.universe.universeConfigTemplate.data);
     }
     // If nodeInstanceList changes, fetch number of available nodes
-    if (
-      getPromiseState(nodeInstanceList).isSuccess() &&
-      getPromiseState(this.props.cloud.nodeInstanceList).isLoading()
-    ) {
-      let numNodesAvailable = nodeInstanceList.data.reduce((acc, val) => {
+    if (getPromiseState(nodeInstanceList).isSuccess()) {
+      let nodesPerRegion = [];
+      nodeInstanceList.data.forEach((node) => {
+        isNonEmptyArray(this.state.regionList) &&
+          this.state.regionList.forEach((region) => {
```

Comment:
seems `isNonEmptyArray()` is exessive as for an empty array `forEach()` won't perform an iteration anyway

Reply:
once you select a region for the primary cluster and hit the `configure read replica` button then the region list is undefined so to avoid that I've used `isNonEmptyArray()`.